sklearn/lda.py

```diff
@@ -450,13 +450,13 @@ def transform(self, X):
         X_new : array, shape (n_samples, n_components)
             Transformed data.
         """
-        check_is_fitted(self, ['xbar_', 'scalings_'], all_or_any=any)
-
-        X = check_array(X)
         if self.solver == 'lsqr':
             raise NotImplementedError("transform not implemented for 'lsqr' "
                                       "solver (use 'svd' or 'eigen').")
-        elif self.solver == 'svd':
+        check_is_fitted(self, ['xbar_', 'scalings_'], all_or_any=any)
```

Comment:
I didn't write these lines, but I guess if the solver is ``eigen`` you only have ``scalings`` and if ``solver="svd"`` you have both. I could remove ``xbar_`` from it, but I don't feel that would make it better or worse. I could branch the check on the ``solver`` parameter but I'm not sure how helpful that would be.